### DISCUSSION TITLE
running tests in the command line

### DISCUSSION THREAD

**Discussion by reymalahay** [bxparks/AUnit#109] [Q&A]
hi there,

is there a way to run `aunit` tests in the command line, before the code under test is loaded to a board?

please advise.

thanks in advance,
rey malahay

**Comment by bxparks** [comment]
You mean something like this?
https://github.com/bxparks/AUnit#epoxyduino

**Comment by reymalahay** [comment]
@bxparks thank you for the response.

not really. i am looking for something similar to #83. my only concern with #83 is that the test code is mixed with the `sut` (system under test)

#83 is the closest to what i want, barring the mixing of code with the `sut`.

what i really want is to write tests, mock out calls like `analogRead` and `digitalRead`, and run these tests in the command line.

is that possible with `aunit`?

please advise.

thanks in advance,
rey malahay



> You mean something like this? https://github.com/bxparks/AUnit#epoxyduino

**Reply by bxparks** [reply]
I get the impression that you are conflating 2 independent things:

1) The ability to mock out certain Arduino API functions that were not designed to be mocked, and,
2) The ability to run AUnit tests on the command line.

The discussion in #83 (and a fair bit of #108 ) relate to the first item.

The reference I gave with https://github.com/bxparks/AUnit#epoxyduino deals with the second.

**Comment by reymalahay** [comment]
Hi @bxparks,

thank you for the response.

> I get the impression that you are conflating 2 independent things:
> 
> 1. The ability to mock out certain Arduino API functions that were not designed to be mocked, and,

no worries. 

> 2. The ability to run AUnit tests on the command line.
> 
> The discussion in #83 (and a fair bit of #108 ) relate to the first item.


thanks, I'll take a look.

> 
> The reference I gave with https://github.com/bxparks/AUnit#epoxyduino deals with the second.


thanks again for this.

kind regards,
rey malahay

**Comment by bxparks** [comment]
(Reopening. I prefer to keep Discussions open so that they are more easily searchable by other people. I close them only when they are irrelevant or not useful.)
